libs/patchwheel/patch.go

```diff
@@ -0,0 +1,279 @@
+package patchwheel
+
+import (
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/databricks/cli/libs/log"
+)
+
+var versionKey []byte = []byte("Version:")
+
+// findFiles returns a slice with a *zip.File for every filename in the arguments slice.
+// The order of the return value matches the order of the arguments.
+// If not found, the corresponding entry is nil.
+func findFiles(r *zip.ReadCloser, files ...string) []*zip.File {
```

Comment:
can you return a map[string]*zip.File instead?

Then the rest of the code is way easier to understand, I really struggled to figure out what's going on.

Or if you need a list in the end you can still return it

```
for _, f := range files {
    filesMap[f] = nil
}

for _, f := range r.File {
	if _, ok := filesMap[f.Name]; ok {
		filesMap[f.Name] = f
	}
}

result := make([]*zip.File, 0, len(filesMap))
for i, f := range files {
    result[i] = filesMap[f]
}

return result

```